gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -449,13 +462,28 @@ private void createJobPropsForTopics(List<String> topics) {
     List<String> list = this.state.getPropAsList(COMPACTION_NORMAL_PRIORITY_TOPICS, StringUtils.EMPTY);
     return DatasetFilterUtils.getPatternsFromStrings(list);
   }
+  
+  private Map<String, Double> getTopicsAndRecompactThreshold() {
+    Map<String, Double> topicsAndRecompactThreshold = Maps.newHashMap();
+    for (String entry : this.state.getPropAsList(COMPACTION_LATEDATA_THRESHOLD_FOR_RECOMPACT_PER_TOPIC, StringUtils.EMPTY)) {
+      String[] topicAndRecompactThreshold = Iterables.toArray(Splitter.on(TOPIC_AND_RECOMPACT_THRESHOLD_SEPARATOR)
```

Comment:
This should support regex, and multiple topics with the same threshold. The format could look like this: `A.*,B.*:0.2 C.*,D.*:0.3`